### ISSUE TITLE
DragNDrop Upload Plugin's custom upload function not working.

### ISSUE THREAD

**Issue by fdemian** [draft-js-plugins/draft-js-plugins#2518]
Hi everyone. 
First of all, thanks for the plugins, they are awesome 😄 . 

## Environment

Plugins used: 

  - @draft-js-plugins/alignment: "^5.0.2",
  - @draft-js-plugins/drag-n-drop: "^4.2.1",
  - @draft-js-plugins/drag-n-drop-upload: "^4.1.1",
  - @draft-js-plugins/editor: "^4.1.0",
  - @draft-js-plugins/focus: "^4.1.2",
  - @draft-js-plugins/image: "^4.1.2",
  - @draft-js-plugins/resizeable: "^5.0.2",
  -  draft-js: "0.11.7"

## Description

I’m using the drag-n-drop image uploader. Everything seems to work fine with the code posted on the Draft-Plugins page, except that custom upload behavior seems to be failing. 

My goal is to actually upload the images to the server, then show them in the editor instead of the current behavior (show them in the editor without uploading them to the server). 

According to the documentation (on the [Alignment + Resize + Focus + Drag'n'Drop + Drag'n'Drop Upload Example]( https://www.draft-js-plugins.com/plugin/image)) this should be possible.

This is the code that I’m currently using (there is a Codesandbox live example below) :

https://github.com/fdemian/draft-plugins-test/blob/master/src/Editor.js

However, the mock upload function does not seem to get called. 
I dug around a bit and it seems that the relevant code in the drag-n-drop upload component is commented out.

https://github.com/draft-js-plugins/draft-js-plugins/blob/master/packages/drag-n-drop-upload/src/handleDroppedFiles.ts#L56-L112

It seems like without that code the upload function will not work. The handleDroppedFiles function creates a new form data object and populates it with files, but does not seem to be calling the custom upload function.

## Reproducible Demo

Sandbox: 
https://codesandbox.io/s/damp-haze-2srcj

### Repro steps: 

1. Open the sandbox.
2. Drag and drop an image into the editor. 

**Expected result**: 
The custom upload function is called, it should print a few statements in the console and launch an alert popup. 

**Actual result**: 
The image is uploaded as a b64 file and the  custom upload function is not called.

**Comment by fxOne** [comment]
Duplicated to #2150.
It would be nice if you could fix issue as you already looked into it.

**Comment by fdemian** [comment]
Sure, I have no problem submiting a PR.  
Is there any context on why the commented out code was commented in the first place?

**Comment by fxOne** [comment]
I have no idea. It was already comment out when I took care about this project.

**Comment by fdemian** [comment]
Status update: 

I've managed to get [the code](https://github.com/fdemian/draft-js-plugins/blob/master/packages/drag-n-drop-upload/src/handleDroppedFiles.ts) to the point where it can upload an image (or at least, simulate uploading one in lieu of an actual server) and insert it in the editor. The code needs more love though.

I suspect the mockUpload functionality never actually worked and/or was never in an official released. There were several functions in the upload code that were missing, or components inside the code that aren't being called (like the [Upload Placeholder](https://github.com/draft-js-plugins/draft-js-plugins/blob/master/packages/drag-n-drop-upload/src/components/UploadPlaceholder.tsx) - I suspect it was suppossed to be inserted in place of the base 64 encoded image). 

TL;DR: I'm making progress but this probably needs more work (and documentation).

**Comment by fdemian** [comment]
[PR #2534](https://github.com/draft-js-plugins/draft-js-plugins/pull/2534) should solve this.
